docs/book/book.typ

```diff
@@ -0,0 +1,178 @@
+
+A diagram representing the relationships and hierarchies between several key classes in the architecture. The stack shown on the diagram only covers Clouseau layer and facade layer.
+
+#box(services-inheritance-diagram)
+
+#pagebreak()
+
+= Index lifecycle management
+
+This index lifecycle management diagram illustrates the interaction between Dreyfus and Clouseau during index lifecycle management, focusing on how ETS tables and LRU cache coordinate with monitors and links. On the Dreyfus side, dreyfus_index_manager maintains two ETS tables: BY_INDEX, which maps {Db, Index} to process IDs, and BY_PID, which tracks active processes for cleanup and routing. When an index is requested, dreyfus_index_manager either retrieves the PID from ETS or calls clouseau_rpc:open_index/3 to forward the request to Clouseau. On the Clouseau side, IndexManagerService checks the OpenIndexLRU cache for an existing index actor; if absent, it spawns a new IndexService actor and adds it to the LRU. Each opened index is represented by two actors—dreyfus_index in Erlang and IndexService in Scala—linked together so that termination of one propagates to the other. Monitors are established on both sides: dreyfus_index_manager monitors dreyfus_index, and IndexManagerService monitors IndexService. This design ensures consistency: if dreyfus_index dies, the manager receives a DOWN signal and removes the entry from ETS, while the link triggers an EXIT signal to terminate IndexService, which in turn is removed from the LRU. Conversely, if IndexService dies, IndexManagerService cleans up the LRU entry, and the link causes dreyfus_index to exit, prompting ETS cleanup. This bidirectional monitoring and linking strategy guarantees that stale references are eliminated promptly, maintaining system integrity across distributed components.
```

Comment:
I would recommend putting names like `BY_INDEX` or `BY_PID` in backticks for verbatim formatting.

Reply:
You would be surprised but by default the backtics style is the same as the rest of the text. You would have to give a `#show` directive in the beginning of the file to make it look different.  Something like:

```
#show raw.where(block: false): set text(font: "Fira Code", size: 9pt)
#show raw.where(block: true): set text(font: "Cascadia Code", size: 10pt)   
```